### PR TITLE
Task1 done

### PR DESCRIPTION
**Pull request by Skeep747**

### CODE REVIEW

File: Airplane-tickets-UA.Data/Airport.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Airplane_tickets_UA.Data
+{
+    public class Airport
+    {
+        public int Id { get; set; }
+        public ICAO ICAO { get; set; }
+        public string Name { get; set; }
+
+        public int CityId { get; set; }
```
Comment (budda2) [inline Airplane-tickets-UA.Data/Airport.cs:9]: i guess there is no need to store both city Id and City itself, choose one from them

File: Airplane-tickets-UA.Data/City.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Airplane_tickets_UA.Data
+{
+    public class City
+    {
+        public int Id { get; set; }
```
Comment (budda2) [inline Airplane-tickets-UA.Data/City.cs:8]: no need to have both int ang GUID Ids, please consider removing one of them

File: Airplane-tickets-UA.Shared/FlightHelper.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    public class FlightHelper
+    {
+        private readonly decimal _discound = 0.85M;
+        private readonly decimal _discound60 = 0.90M;
+        private readonly decimal _discound30 = 0.95M;
+        private readonly decimal _discound1 = 0.50M;
+
+        private readonly Random random = new Random();
+
+        public delegate void FlightGeneratorEventHandler();
```
Comment (budda2) [inline Airplane-tickets-UA.Shared/FlightHelper.cs]: better to have delagates declared in a separate file

**Comment by budda2** [comment]
Overall note: you do have in most classes duplications, w.g FlieghtId, flieght properties/fields, consider choosing only one of them that will remain in class.
One more note: i did see that you fire events, and this is great, the question is: who is responsible for handling them? i haven't seen any subscriptions and event handlings, please add them for your homework to be completed

File: Airplane-tickets-UA.Data/Airport.cs (hunk 1 above)
Comment (Skeep747) [reply Airplane-tickets-UA.Data/Airport.cs:9]: I need both properties for further database samples. "CityId" is convenient for search queries, and "City" is for displaying data.

File: Airplane-tickets-UA.Data/City.cs (hunk 2 above)
Comment (Skeep747) [reply Airplane-tickets-UA.Data/City.cs:8]: I need the ID only for the database, and GUID for displaying in the UI. GUID will be unique, assigned to the flight, and the ID can change depending on database operations.

File: Airplane-tickets-UA.Data/Airport.cs (hunk 1 above)
Comment (budda2) [reply Airplane-tickets-UA.Data/Airport.cs:9]: but you could easily get one having another or at least have one of them private

**Comment by Skeep747** [comment]
Added project worker. He will be responsible for all the events.

File: Airplane-tickets-UA.Shared/FlightHelper.cs [hunk 4]
```diff
@@ -71,10 +67,9 @@ private List<ExistingFlight> GetSpecialOffer(List<ExistingFlight> existingFlight
                         existingFlight.Price *= _discound1;
                         break;
                 }
+                DiscountActivatedEvent?.Invoke(existingFlight.From, existingFlight.To);
```
Comment (budda2) [inline Airplane-tickets-UA.Shared/FlightHelper.cs:70]: yea, now much better, requirement of someone to handle your notification forced you tu pass some data, and that is great, that is what usually happens, notification comes with data explaining how system state has changed.

File: Airplane-tickets-UA.Worker/Program.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+namespace Airplane_tickets_UA.Worker
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
```
Comment (budda2) [inline Airplane-tickets-UA.Worker/Program.cs:17]: Am I right this trick comes from ASP .Net Core? looks really nice, could this be configured to handle remote requests?

**Comment by budda2** [comment]
Review done, looks nice, feel free to merge yourself

**Pull request merged**